src/mon/MonClient.h

```diff
@@ -213,14 +213,19 @@ class MonClient : public Dispatcher {
   void handle_subscribe_ack(MMonSubscribeAck* m);
 
   bool _sub_want(string what, version_t start, unsigned flags) {
-    if ((sub_new.count(what) == 0 &&
-	 sub_sent.count(what) &&
-	 sub_sent[what].start == start &&
-	 sub_sent[what].flags == flags) ||
-	(sub_new.count(what) &&
-	 sub_new[what].start == start &&
-	 sub_new[what].flags == flags))
+    auto sub_new_iter = sub_new.find(what);
```

Comment:
> What's wrong with adding iter to name of the container? Doesn't it make very explicit what variable is doing?

for example, if we want to find an item in a `vector<int>`, one could put
```c++
auto vector_iter = haystack_vector.find(straw_i);
```
or he/she could use
```c++
auto found = v.find(straw);
```

nothing wrong with `sub_new_iter`. but by looking at the (exaggerated) example, probably you can see the difference.

> If I rename sub_new_iter to sub, how should I name sub_sent_iter?

you don't need a new variable name, since it will also be a "sub", so why not just reuse it? like:
```c++
sub = sub_sent.find(what);
```